modules/porous_flow/src/materials/PorousFlowMaterial.C

```diff
@@ -37,6 +37,21 @@ PorousFlowMaterial::PorousFlowMaterial(const InputParameters & parameters)
 {
 }
 
+void
+PorousFlowMaterial::initialSetup()
+{
+  for (auto prop_name : getSuppliedItems())
+  {
+    mooseAssert(_material_data->getMaterialPropertyStorage().hasProperty(prop_name),
```

Comment:
Again me being curious: is it even possible to trigger this Assert?  Do we have a test for it?  I think i probably wrote it initially.... so long ago!

Reply:
I can't think of a way or how to even test it (I just copied and pasted it from `sizeNodalPropery`!

Reply:
Don't worry about testing an assert. Many of them are not testable. They should be viewed as an explicit statement of your assumptions (e.g. I'm assuming that this is true because my code will likely blow up if it's not!). They get completely removed in OPT and OPROF modes so they have zero impact on performance when it matters. However, when you run into trouble you can track down breakdown in your assumptions just by compiling and running in DBG or DEVEL modes. Be prolific with your asserts!